pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -619,23 +619,18 @@ func snapshot(
 	}, nil
 }
 
-// append the given entries to the raft log. Takes the previous values of
-// r.mu.lastIndex, r.mu.lastTerm, and r.mu.raftLogSize, and returns new values.
-// We do this rather than modifying them directly because these modifications
-// need to be atomic with the commit of the batch. This method requires that
-// r.raftMu is held.
+// logAppend adds the given entries to the raft log. Takes the previous
+// lastIndex, lastTerm, raftLogSize, and returns new values. It's the
+// caller's responsibility to maintain exclusive access to the raft log
+// for the duration of the method call.
 //
-// append is intentionally oblivious to the existence of sideloaded proposals.
-// They are managed by the caller, including cleaning up obsolete on-disk
-// payloads in case the log tail is replaced.
-//
-// NOTE: This method takes a engine.Writer because reads are unnecessary when
-// prevLastIndex is 0 and prevLastTerm is invalidLastTerm. In the case where
-// reading is necessary (I.E. entries are getting overwritten or deleted), a
-// engine.ReadWriter must be passed in.
-func (r *Replica) append(
+// logAppend is intentionally oblivious to the existence of sideloaded
+// proposals. They are managed by the caller, including cleaning up obsolete
+// on-disk payloads in case the log tail is replaced.
+func logAppend(
 	ctx context.Context,
-	writer storage.Writer,
+	raftLogPrefix roachpb.Key,
+	rw storage.ReadWriter,
```

Comment:
This now unconditionally takes `ReadWriter`. Were there any conditions under which an "only" `Writer` was passed in. There seems to be only one user of this func, so we're good?